src/main/webapp/app/exam/shared/working-time-update/working-time-control.component.ts

```diff
@@ -0,0 +1,165 @@
+import { Component, Input } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+import { Exam } from 'app/entities/exam.model';
+import { getRelativeWorkingTimeExtension, normalWorkingTime } from 'app/exam/participate/exam.utils';
+import { ArtemisDurationFromSecondsPipe } from 'app/shared/pipes/artemis-duration-from-seconds.pipe';
+import { round } from 'app/shared/util/utils';
+
+@Component({
+    selector: 'jhi-working-time-control',
+    templateUrl: './working-time-control.component.html',
+    styleUrls: ['./working-time-control.component.scss'],
+    providers: [
+        ArtemisDurationFromSecondsPipe,
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: WorkingTimeControlComponent,
+        },
+    ],
+})
+export class WorkingTimeControlComponent implements ControlValueAccessor {
+    // Control disabled state
+    @Input() disabled = false;
+    @Input() showRelative = false;
+
+    @Input() durationLabelText?: string;
+    @Input() relativeLabelText?: string;
+
+    // The exam for which the working time should be updated
+    @Input()
+    set exam(exam: Exam | undefined) {
+        this.currentExam = exam;
+        this.initWorkingTimeFromCurrentExam();
+    }
+
+    get exam(): Exam | undefined {
+        return this.currentExam;
+    }
+
+    private currentExam?: Exam;
+    private touched = false;
+    private onTouched = () => {};
+    private onChange: (_: number) => void = () => {};
+
+    workingTime = {
+        hours: 0,
+        minutes: 0,
+        seconds: 0,
+        percent: 0,
+    };
+
+    constructor(private artemisDurationFromSecondsPipe: ArtemisDurationFromSecondsPipe) {}
+
+    // ControlValueAccessor implementation
```

Comment:
What do you mean by that?

Reply:
I just miss Xcode's `// MARK: -` comment keyword 🥲 just some helper for me during implementation what part belongs to the impl of the interface, will drop that.

```suggestion
```